src/main.rs

```diff
@@ -10,6 +10,9 @@ extern crate test;
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
+#[cfg(all(feature = "ring", feature = "aws-lc-rs"))]
+compile_error! {"Feature \"ring\" and  feature \"aws-lc-rs\" cannot be enabled at the same time."}
```

Comment:
Is this something we have to do or does rustls/hickory already do it for us?

Also, do we even need this? It looks like the way the code is written right now, aws-lc would overwrite but not conflict ring.

Reply:
In this case (since crab-hole is a binary) you are right, there is not really a point in having both enabled.

However, when you write a library, let's call it _a_, and some other library, let's call it _b_, uses library _a_, it can be useful to have features that "additionally" "disable" the default for _a_ so that eventhough library _b_ forces the default features of library _a_, you can have the alternative behaviour.